src/t8_forest/t8_forest_adapt.cxx

```diff
@@ -22,14 +22,197 @@
+ * \return                   The index of the first family member whose family is
+ *                           defined by \a telements_pos in \a telement.                    
+ * \note The element with index \a telements_pos must be the last child in its family.
+ *       \see t8_forest_adapt_coarsen_recursive.
+ * \note If the element with index \a telements_pos in \a telement can not be coarsened
+ *       recursively, return INT32_MIN.
+ */
+static t8_locidx_t
+t8_forest_pos (t8_forest_t forest,
+               t8_eclass_scheme_c *ts,
+               t8_element_array_t *telements,
+               t8_locidx_t telements_pos)
+{
+  t8_element_t       *element; /* element with index telements_pos */
+  t8_element_t       *element_compare;
+  t8_element_t       *element_parent;
+  t8_element_t       *element_parent_compare;
+  int                 level_current;
+  int                 level_compare;
+  t8_locidx_t         child_id;
+  t8_locidx_t         pos;
+  t8_locidx_t         el_iter; /* Loop running variable */
+  t8_locidx_t         elements_in_array;
+  t8_locidx_t         num_siblings;
+
+  pos = telements_pos;
+  elements_in_array = t8_element_array_get_count (telements);
+  T8_ASSERT (0 <= pos && pos < elements_in_array);
+  element = t8_element_array_index_locidx (telements, pos);
+  element_compare = t8_element_array_index_locidx (telements, pos);
+  level_current = ts->t8_element_level (element);
+  num_siblings = ts->t8_element_num_siblings (element);
+  child_id = ts->t8_element_child_id (element);
+  
+  /* Elements whose family consist of exactly one element do not 
+   * get coarsened recursively.
+   * If child_id is not last, elements cannot be coarsened recursively.
+   * Elements with level 0 cannot be further coarsened. */
+  if (!(child_id > 0 && child_id == num_siblings - 1) || level_current == 0) {
+    return INT32_MIN;
+  }
+  /* If the forest is complete, the family is also complete. 
+   * Thus, the index of the first member can be determined. */
+  if (!forest->is_incomplete) {
+    return pos - num_siblings - 1;
+  }
+  
+  T8_ASSERT (child_id > 0 && child_id == num_siblings - 1);
+  T8_ASSERT (level_current > 0);
+
+  ts->t8_element_new (1, &element_parent_compare);
+  ts->t8_element_new (1, &element_parent);
+  /* Get parent of a family member by coarsening last member. */
+  ts->t8_element_parent (element, element_parent);
+
+  /* Loop backward over all possible familie members until we hit an 
+   * element that is not part of the family or we have reached the 
+   * maximum number of member. */
+  for (el_iter = 1; el_iter < num_siblings 
+                    && el_iter < elements_in_array; el_iter++) {
+    pos = telements_pos - el_iter;
+    T8_ASSERT (0 <= pos && pos < elements_in_array);
+    element_compare = t8_element_array_index_locidx (telements, pos);
+    /* Quick check by level. Not mandatory. */
+    level_compare = ts->t8_element_level (element_compare);
```

Comment:
The test is not complex and is intended to prevent parent elements from being calculated and compared too frequently.
In this way, computing time can probably be saved in most cases.
I will extend the documentation to make this clearer.